src/common/options/global.yaml.in

```diff
@@ -1189,6 +1189,34 @@ options:
   fmt_desc: The maximum port number to which an OSD or MDS daemon will bind.
   default: 7568
   with_legacy: true
+- name: ms_bind_port_min_osd
```

Comment:
Are those new options really necessary? Is there an advantage over setting e.g. `ms_bind_port_min` differently in the OSD's and MDS' sections? To illustrate the idea, I'm pasting parts of the `vstart.sh`-generated `ceph.conf`:

```
[mds]

        log file = /home/rzarzynski/ceph1/build/out/$name.log
        admin socket = /home/rzarzynski/ceph1/build/asok/$name.asok
        chdir = ""
        pid file = /home/rzarzynski/ceph1/build/out/$name.pid
        heartbeat file = /home/rzarzynski/ceph1/build/out/$name.heartbeat
# ...
[osd]

        log file = /home/rzarzynski/ceph1/build/out/$name.log
        admin socket = /home/rzarzynski/ceph1/build/asok/$name.asok
```

Reply:
Yeah, these shouldn't be necessary.  If existing `ms_bind_port_min` and `ms_bind_port_max` options don't work when placed into an appropriate section, we should fix that instead of adding new options.  (Also, adding new options just for OSD and MDS daemons misses the manager daemon).

Reply:
> Yeah, these shouldn't be necessary. If existing `ms_bind_port_min` and `ms_bind_port_max` options don't work when placed into an appropriate section, we should fix that instead of adding new options. (Also, adding new options just for OSD and MDS daemons misses e.g. the manager daemon).

Note that `ceph.conf` file syntax and `ceph config set` command allow for setting options per cephx entity (i.e. per daemon), e.g. `osd.0` or `mds.b`.  One isn't limited to just "all OSDs" or "all MDSs".

Reply:
I hadn't even thought of this. Are ms_bind_port_min and ms_bind_port_max applicable to each daemon? I've been thinking about how to fix it for a few weeks, and then I've come to a code fix. There was an easy way, but I couldn't find it. In fact, when I run it that way, it works. Why didn't I think of this method? Haha, my first PR attempt failed, but I'm glad I got a cleaner setup.

PR will be closed. thank you